_docs/tasks/traffic-management/mirroring.md

```diff
@@ -0,0 +1,228 @@
+
+sleep service:
+
+ ```bash
+cat <<EOF | kubectl create -f -
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: sleep
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: sleep
+    spec:
+      containers:
+      - name: sleep
+        image: tutum/curl
+        command: ["/bin/sleep","infinity"]
+        imagePullPolicy: IfNotPresent
+EOF
+```   
+
+
+
+
+## Mirroring
+
+Let's set up a scenario to demonstrate the traffic-mirroring capabilities of Istio. We have two versions of our `httpbin` service. By default Kubernetes will load balance across both versions of the service. We'll use Istio to force all traffic to v1 of the `httpbin` service. 
+
+### Creating default routing policy
+
+1. Create a default route rule to route all traffic to `v1` of our `httpbin` service:
+
+```bash
+cat <<EOF | istioctl create -f -
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: httpbin-default-v1
+spec:
+  destination:
+    name: httpbin
+  precedence: 5
+  route:
+  - labels:
+      version: v1
+EOF
+```
+
+Now all traffic should go to `httpbin v1` service. Let's try sending in some traffic:
+
+```bash
+export SLEEP_POD=$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})
+kubectl exec -it $SLEEP_POD -c sleep -- sh -c 'curl  http://httpbin:8080/headers'
+
+{
+  "headers": {
+    "Accept": "*/*", 
+    "Content-Length": "0", 
+    "Host": "httpbin:8080", 
+    "User-Agent": "curl/7.35.0", 
+    "X-B3-Sampled": "1", 
+    "X-B3-Spanid": "eca3d7ed8f2e6a0a", 
+    "X-B3-Traceid": "eca3d7ed8f2e6a0a", 
+    "X-Ot-Span-Context": "eca3d7ed8f2e6a0a;eca3d7ed8f2e6a0a;0000000000000000"
+  }
+}
+```
+
+If we check the logs for `v1` and `v2` of our `httpbin` pods, we should see access log entries for only `v1`:
+
+```bash
+$  kubectl logs -f httpbin-v1-2113278084-98whj -c httpbin 
+127.0.0.1 - - [07/Feb/2018:00:07:39 +0000] "GET /headers HTTP/1.1" 200 349 "-" "curl/7.35.0" 
+```
+
+2. Create a route rule to mirror traffic to v2
+
+```bash
+cat <<EOF | istioctl create -f -
+apiVersion: config.istio.io/v1alpha2
+kind: RouteRule
+metadata:
+  name: mirror-traffic-to-httbin-v2
+spec:
+  destination:
+    name: httpbin
+  precedence: 11
+  route:
+  - labels:
+      version: v1
+    weight: 100
+  - labels: 
+      version: v2
+    weight: 0
+  mirror:
+    name: httpbin
+    labels:
+      version: v2
+EOF
+```
+
+This route rule specifies we route 100% of the traffic to v1 and 0% to v2. At the moment, it's necesary to call out the v2 service explicitly because this is what creates the envoy-cluster definitions in the background. The last stanza specifies we want to mirror to the `httpbin v2` service.
+
+
+Now if we send in traffic:
+
+```bash
+kubectl exec -it $SLEEP_POD -c sleep -- sh -c 'curl  http://httpbin:8080/headers'
+```
+
+We should see access logging for both `v1` and `v2`. The access logs created in `v2` is the mirrored requests that are actually going to `v1`.
+   
```

Comment:
Maybe it's obvious, but it might we worth mentioning that the response from the mirror is discarded.

Reply:
Good point. will add that.